compiler/semantic/expr.go

```diff
@@ -116,6 +116,42 @@ func (a *analyzer) semExpr(e ast.Expr) (dag.Expr, error) {
 			Name: "cast",
 			Args: []dag.Expr{expr, typ},
 		}, nil
+	case *ast.IndexExpr:
+		expr, err := a.semExpr(e.Expr)
+		if err != nil {
+			return nil, err
+		}
+		// XXX This should be type checked as either an int, string or var.
```

Comment:
I think we have to allow any type here since a map key can have any type.
```suggestion
```

Reply:
Good point!